tests/acceptance/blocks/filterboxcomponent.js

```diff
@@ -0,0 +1,38 @@
+import { Selector, t } from 'testcafe';
+import FilterOptionsComponentBlock from './filteroptionscomponent';
+/**
+ * This class models user interactions with the {@link FilterBoxComponent}.
+ */
+export default class FilterBoxComponentBlock {
+  constructor (selector) {
+    this._container = '.yxt-FilterBox-container';
+    this._selector = selector || Selector(this._container);
+  }
+
+  /**
+   * Gets the child FilterOptions block with the given title.
+   * @param {String} title
+   */
+  async getFilterOptions (title) {
+    const filterOptions = await this._selector.find('.yxt-FilterOptions-fieldSet').withText(title);
+    return new FilterOptionsComponentBlock(filterOptions);
+  }
+
+  /**
+   * Apply the filters in this filter box.
+   */
+  async applyFilters () {
+    const applyButton = await this._selector.find('.js-yext-filterbox-apply');
+    await t.click(applyButton);
+    // TODO more sophisticated waiting mechanism, docs suggest waiting for a specific DOM change
+    await t.wait(1000);
```

Comment:
Let's add the improved waiting mechanism now. Waiting for a specific amount of time can lead to flaky/unpredictable tests.

Reply:
First tried to use RequestLogger, but apparently testcafe already has built-in waiting for xhr requests. Removed this t.wait and wasn't seeing the waiting issues I was having before, so just removed it.